pkg/pointerdb/pointerdb_test.go

```diff
@@ -143,92 +147,178 @@ func TestServiceDelete(t *testing.T) {
 }
 
 func TestServiceList(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
+	db := teststore.New()
+	server := Server{DB: db, logger: zap.NewNop()}
 
-	keys := storage.Keys{
-		storage.Key(paths.New("sample.jpg").Bytes()),
-		storage.Key(paths.New("music").Bytes()),
-		storage.Key(paths.New("music/song1.mp3").Bytes()),
-		storage.Key(paths.New("music/song2.mp3").Bytes()),
-		storage.Key(paths.New("music/album/song3.mp3").Bytes()),
-		storage.Key(paths.New("music/song4.mp3").Bytes()),
-		storage.Key(paths.New("videos").Bytes()),
-		storage.Key(paths.New("videos/movie.mkv").Bytes()),
+	key := func(s string) storage.Key {
+		return storage.Key(paths.New(s).Bytes())
 	}
 
-	for i, tt := range []struct {
-		prefix       string
-		startAfter   string
-		endBefore    string
-		recursive    bool
-		limit        int
-		metaFlags    uint32
-		apiKey       []byte
-		returnedKeys storage.Keys
-		expectedKeys storage.Keys
-		expectedMore bool
-		err          error
-		errString    string
-	}{
-		{"", "", "", true, 0, meta.None, nil, keys, keys, false, nil, ""},
-		{"", "", "", true, 0, meta.All, nil, keys, keys, false, nil, ""},
-		{"", "", "", true, 0, meta.None, []byte("wrong key"), keys, keys, false,
-			nil, status.Errorf(codes.Unauthenticated, "Invalid API credential").Error()},
-		{"", "", "", true, 0, meta.None, nil, keys, keys, false,
-			errors.New("list error"), status.Errorf(codes.Internal, "list error").Error()},
-		{"", "", "", true, 2, meta.None, nil, keys, keys[:2], true, nil, ""},
-		{"", "", "", false, 0, meta.None, nil, keys, storage.Keys{keys[0], keys[1], keys[6]}, false, nil, ""},
-		{"", "", "videos", false, 0, meta.None, nil, keys, keys[:2], false, nil, ""},
-		{"music", "", "", false, 0, meta.None, nil, keys[2:], storage.Keys{keys[2], keys[3], keys[5]}, false, nil, ""},
-		{"music", "", "", true, 0, meta.None, nil, keys[2:], keys[2:6], false, nil, ""},
-		{"music", "song1.mp3", "", true, 0, meta.None, nil, keys, keys[3:6], false, nil, ""},
-		{"music", "song1.mp3", "album/song3.mp3", true, 0, meta.None, nil, keys, keys[3:4], false, nil, ""},
-		{"music", "", "song4.mp3", true, 0, meta.None, nil, keys, keys[2:5], false, nil, ""},
-		{"music", "", "song4.mp3", true, 1, meta.None, nil, keys, keys[4:5], true, nil, ""},
-		{"music", "", "song4.mp3", false, 0, meta.None, nil, keys, keys[2:4], false, nil, ""},
-		{"music", "song2.mp3", "song4.mp3", true, 0, meta.None, nil, keys, keys[4:5], false, nil, ""},
-		{"mus", "", "", true, 0, meta.None, nil, keys[1:], nil, false, nil, ""},
-	} {
-		errTag := fmt.Sprintf("Test case #%d", i)
+	pointer := &pb.Pointer{}
+	pointer.CreationDate = ptypes.TimestampNow()
 
-		db := NewMockKeyValueStore(ctrl)
-		s := Server{DB: db, logger: zap.NewNop()}
+	pointerBytes, err := proto.Marshal(pointer)
+	if err != nil {
+		t.Fatal(err)
+	}
+	pointerValue := storage.Value(pointerBytes)
+
+	err = storage.PutAll(db, []storage.ListItem{
```

Comment:
this seems ridiculous, i know, should we test for special characters? empty spaces?

Reply:
I'll change few of the entries to use unicode, spaces and some other characters to cover them better.